subsys/net/ip/net_if.c

```diff
@@ -555,6 +561,90 @@ static void dad_timeout(struct k_work *work)
 		 */
 		net_ipv6_nbr_rm(iface, &ifaddr->address.in6_addr);
 	}
+
+	ifaddr->dad.timer_timeout = 0;
+
+	sys_slist_find_and_remove(&active_dad_timers, &ifaddr->dad.node);
+}
+
+static bool dad_check_timeout(s64_t start, u32_t time, s64_t timeout)
+{
+	start += time;
+	start = abs(start);
```

Comment:
Using signed 64bit numbers for always-increasing time is indeed confusing...

Reply:
The k_uptime_get() returns s64_t, I am not sure what is the point in your comment here.

Reply:
k_uptime_get() returns s64_t, but is described as "This routine returns the elapsed time since the system booted". It's clear that elapsed time goes only forward, and cannot be negative. At one point in the unattainable future it will overflow, but C standard says that the result of signed overflow is undefined, so a good C programmer cannot do anything about it anyway. But maybe you personally don't care about the C standard, and assume that signed numbers are 2's-complement. But using abs() in this case doesn't make any sense, because at the point of overflow, such abs'ed signed time will go backwards. Let me demonstrate this on 16-bit numbers:

After 32768, there will be -32768. Applying abs() to it will lead to the same number. But then there will be -32767, which abs will turn into 32767, etc. I.e. the "time" on which you perform calculations will go as follows:

32766, 32767, -32768, 32767, 32766, 32765, ... 10, 9, 8, ...

So, using, abs() totally doesn't make sense. And that's after performing signed operations on wrap-around time, which don't make sense first.

Reply:
Your comments are in weird order in github and this gets quite convoluted.

Anyway, about the conversions. We are only interested in difference of the numbers and at least gcc returns the difference correctly as shown by this example

```
main()
{
	int32_t max_int = INT32_MAX;
	int32_t max_int2 = INT32_MAX + 2;
	int32_t diff = max_int - max_int2;

	printf("max sint %d\n", max_int);
	printf("max + 2 sint %d\n", max_int2);
	printf("diff %d\n", diff);
}
```
will print
```
max sint 2147483647
max + 2 sint -2147483647
diff -2
```

Taking abs() from the diff makes sense here.